convert2rhel/unit_tests/backup_test.py

```diff
@@ -398,3 +399,253 @@ def test_restorable_package_backup(pretend_os, is_eus_system, has_internet_acces
+        assert popped_restorable3 is restorable3
+
+        assert restorable1.called["restore"] == 1
+        assert restorable2.called["restore"] == 1
+        assert restorable3.called["restore"] == 1
+
+    def test_pop_when_empty(self, backup_controller):
+        with pytest.raises(IndexError, match="pop from empty list"):
+            backup_controller.pop()
+
+    def test_pop_all(self, backup_controller):
+        restorable1 = MinimalRestorable()
+        restorable2 = MinimalRestorable()
+        restorable3 = MinimalRestorable()
+
+        backup_controller.push(restorable1)
+        backup_controller.push(restorable2)
+        backup_controller.push(restorable3)
+
+        restorables = backup_controller.pop_all()
+
+        assert len(restorables) == 3
+        assert restorables[0] is restorable3
+        assert restorables[1] is restorable2
+        assert restorables[2] is restorable1
+
+        assert restorable1.called["restore"] == 1
+        assert restorable2.called["restore"] == 1
+        assert restorable3.called["restore"] == 1
+
+    def test_ready_to_push_after_pop_all(self, backup_controller):
+        restorable1 = MinimalRestorable()
+        restorable2 = MinimalRestorable()
+
+        backup_controller.push(restorable1)
+        popped_restorables = backup_controller.pop_all()
+        backup_controller.push(restorable2)
+
+        assert len(popped_restorables) == 1
+        assert popped_restorables[0] == restorable1
+        assert len(backup_controller._restorables) == 1
+        assert backup_controller._restorables[0] is restorable2
+
+    def test_pop_all_when_empty(self, backup_controller):
+        with pytest.raises(IndexError, match="pop from empty list"):
+            backup_controller.pop_all()
+
+
+@pytest.fixture
+def run_subprocess_with_empty_rpmdb(monkeypatch, tmpdir):
+    """When we use rpm, inject our fake rpmdb instead of the system one."""
+    rpmdb = os.path.join(str(tmpdir), "rpmdb")
+    os.mkdir(rpmdb)
+
+    class RunSubprocessWithEmptyRpmdb(object):
+        def __init__(self):
+            self.called_with = []
+
+        def __call__(self, *args, **kwargs):
+            self.called_with.append((args, kwargs))
+
+            if args[0][0] == "rpm":
+                args[0].extend(["--dbpath", rpmdb])
+
+            return real_run_subprocess(*args, **kwargs)
+
+    real_run_subprocess = utils.run_subprocess
+    instrumented_run_subprocess = RunSubprocessWithEmptyRpmdb()
+    monkeypatch.setattr(utils, "run_subprocess", instrumented_run_subprocess)
+
+    return instrumented_run_subprocess
+
+
+class TestRestorableRpmKey:
+    gpg_key = os.path.realpath(
+        os.path.join(os.path.dirname(__file__), "../data/version-independent/gpg-keys/RPM-GPG-KEY-redhat-release")
+    )
+
+    @pytest.fixture
+    def rpm_key(self):
+        return backup.RestorableRpmKey(self.gpg_key)
+
+    def test_init(self):
+        rpm_key = backup.RestorableRpmKey(self.gpg_key)
```

Comment:
Changing this to use the fixture would be a suitable change? 

```suggestion
    def test_init(self, rpm_key):
```

Reply:
I'm not sure.  Here's the pro and con that I'm struggling with:
* Pro: We don't have to repeat ourselves since the rpm_key fixture does exactly what this code is doing.  If we ever change the parameters to `backup.RestorableRpmKey.__init__()`, we would only have one place to change it, in the fixture.
* Con: This function is specifically for testing `RestorableRpmKey.__init__()`.  Relying on the fixture to call that function seems like we're relying on the implementation of the fixture rather than being sure that we're actually calling the function that we're testing by making that call explicit in our test function.

I think I lean towards making the call explicit here.  Does that sound okay to you?

Reply:
Hey, sorry for the late response! Yes, that sounds totally fine by me.